workshops/NeuralNetFromScratch/README.md

```diff
@@ -0,0 +1,237 @@
+
+<img src="https://hackclub.slack.com/files/U010VMPCEBF/F01DLN09RFS/backprop.png?origin_team=T0266FRGM&origin_channel=D010V8S08G4" alt="Back Propagation"/>
+ 
+ 
+# Training it
+Now let's actually train our model!
+ 
+In the following lines of code, I will create training inputs and transpose them.
+ 
+```python
+trainingInputs = np.array([[0,0,1],
+                            [1,1,1],
+                            [1,0,1],
+                            [0,1,1]])
+ 
+trainingOutputs = np.array([[0,1,1,0]]).T
+```
+ 
+Then I will pass this through the train function:
+```python
+train(trainingInputs, trainingOutputs, 10000)
+```
+ 
+# Get The Results!
+ 
+We will prompt the user for inputs:
+```python
+ 
+input1 = str(input("Input 1: "))
+input2 = str(input("Input 2: "))
+input3 = str(input("Input 3: "))
+```
+ 
+And finally we will pass it through our network!!!
+```python
+base(np.array([input1, input2, input3]))
+```
+ 
+```
+input 1: 1
+input 2: 0
+input 3: 0
+```
+Output:
+Something close to 1, for example `[0.99993704]`
+
+The neural network learns that if the inputs are [1,1,1], [1,0,1], or [1,0,0] that the output is 1, and otherwise 0.
+ 
+<img src="https://hackclub.slack.com/files/U010VMPCEBF/F01DLN2LDHN/screenshot.png?origin_team=T0266FRGM&origin_channel=D010V8S08G4" alt="Final result">
+ 
+ 
+View code here: [https://github.com/JohnLins/NeuralNetwork](https://github.com/JohnLins/NeuralNetwork)
+Run code here: [https://repl.it/@JohnLins/NeuralNetwork](https://repl.it/@JohnLins/NeuralNetwork)
+ 
+# Final code
+```python
+import numpy as np
+
+np.random.seed(1)
+
+synapticWeights = 2 * np.random.random((3, 1)) - 1
+
+def sigmoid(x):
+    return 1 / (1 + np.exp(-x))
+ 
+def sigmoidDerivative(x):
+    return x * (1 - x)
+
+
+def base(inputs):
+    inputs = inputs.astype(float) 
+    return sigmoid(np.dot(inputs, synapticWeights))
+
+def train(trainingInputs, trainingOutputs, trainingIterations):
+    global synapticWeights
+    for iteration in range(trainingIterations):
+        
+        output = base(trainingInputs)
+        
+        error = trainingOutputs - output
+        
+        adjustments = np.dot(trainingInputs.T, error * sigmoidDerivative(output))
+
+        synapticWeights += adjustments
+
+
+print("Random starting weights: ")
+print(synapticWeights)
+
+
+trainingInputs = np.array([[0,0,1],
+                            [1,1,1],
+                            [1,0,1],
+                            [0,1,1]])
+
+trainingOutputs = np.array([[0,1,1,0]]).T
+
+  
+train(trainingInputs, trainingOutputs, 10000)
+
+
+print("Weights after training: ")
+print(synapticWeights)
+print("-------------------")
+
+input1 = str(input("Input 1: "))
+input2 = str(input("Input 2: "))
+input3 = str(input("Input 3: "))
+    
+print("Input data = ", input1, input2, input3)
+print("-------------------")
+print("Output data: ")
+print(base(np.array([input1, input2, input3])))
+```
+
+
+
+# Run
```

Comment:
All workshops have to use some online tool. I recommend using [repl.it](https://repl.it). Instead of including a separate step for running it in a terminal, include a step for running it inside repl.it at the end of the previous section